### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by bjones1**
This contains updates for my literate programming directive:

1. Use celery instead of web2py's unreliable scheduler.
2. Use Python 3 for subprocess calls.

I've also updated the Docker files so it runs correctly there.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,6 +4,8 @@ python:
 - '3.7'
 services:
 - postgresql
+# For Celery (web2py scheduler replacement).
+- redis-server
```
Comment (bnmnetp) [inline .travis.yml]: interesting.... Once this is in place we can move to using redis to manage session data instead of Postgres.

Comment (bjones1) [reply .travis.yml]: Hmmm - interesting. But I don't see any way to do that -- session data gets stored to either a file or a DB table, and DAL doesn't support redis AFAIK. Or am I missing something? (I hope so -- that sounds really neat...)

Comment (bnmnetp) [reply .travis.yml]: http://web2py.com/books/default/chapter/29/13?search=cache#Sessions-in-Redis

Comment (bnmnetp) [reply .travis.yml]: web2py also supports using redis in places where you would also use memcache.

Comment (bjones1) [reply .travis.yml]: Wow, cool! Nice find!

**Comment by bjones1** [comment]
Pinging...

**Comment by bnmnetp** [comment]
Hey, its on my list of things to test, but I’m not going to get to it until next week.

**Comment by bjones1** [comment]
OK, no problem. I was just afraid it had fallen off your list.

**Pull request merged**